Tests/PrettyPrinterTests/ExtensionDeclTests.swift

```diff
@@ -0,0 +1,222 @@
+public class ExtensionDeclTests: PrettyPrintTestCase {
+  public func testBasicExtensionDeclarations() {
+    let input =
+      """
+      extension MyExtension {
+        let A: Int
+        let B: Bool
+      }
+      public extension MyExtension {
+        let A: Int
+        let B: Bool
+      }
+      public extension MyLongerExtension {
+        let A: Int
+        let B: Bool
+      }
+      """
+
+    let expected =
+      """
+      extension MyExtension {
+        let A: Int
+        let B: Bool
+      }
+      public extension MyExtension {
+        let A: Int
+        let B: Bool
+      }
+      public extension
+      MyLongerExtension {
+        let A: Int
+        let B: Bool
+      }
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 33)
+  }
+
+  public func testGenericExtensionDeclarations() {
+    let input =
+      """
+      extension MyExtension<T> {
```

Comment:
Wait, does the compiler actually parse these and give you a valid `GenericArgumentClauseSyntax` in this position? Extensions can't have generic argument clauses (there's not even a slot for it in the list of [`ExtensionDeclSyntax`'s children](https://github.com/apple/swift-syntax/blob/0.40200.0/Sources/SwiftSyntax/gyb_generated/SyntaxNodes.swift#L6129-L6136) so... how does this work? 😨

Reply:
GenericArgumentClause is being populated by the extension in the AST. Actually, the compiler has let though some other things that clearly weren't right:
```
public indirect MyEnum {
  ...
```
for example.